symfony/plugins/orangehrmRecruitmentPlugin/Api/VacancyAPI.php

```diff
@@ -0,0 +1,387 @@
+
+
+use DateTime;
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\CrudEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Exception\RecordNotFoundException;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Serializer\NormalizeException;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Exception\DaoException;
+use OrangeHRM\Entity\Vacancy;
+use OrangeHRM\Pim\Dto\EmployeeSearchFilterParams;
+use OrangeHRM\Recruitment\Api\Model\VacancyModel;
+use OrangeHRM\Recruitment\Dto\VacancySearchFilterParams;
+use OrangeHRM\Recruitment\Service\VacancyService;
+
+class VacancyAPI extends Endpoint implements CrudEndpoint
+{
+    public const FILTER_JOB_TITLE_ID = 'jobTitleId';
+    public const FILTER_VACANCY_ID = 'vacancyId';
+    public const FILTER_HIRING_MANAGER_ID = 'hiringManagerId';
+    public const FILTER_STATUS = 'status';
+
+    public const PARAMETER_NAME = 'name';
+    public const PARAMETER_DESCRIPTION = 'description';
+    public const PARAMETER_NUM_OF_POSITIONS = 'numOfPositions';
+    public const PARAMETER_STATUS = 'status';
+    public const PARAMETER_IS_PUBLISHED = 'isPublished';
+    public const PARAMETER_JOB_TITLE_ID = 'jobTitleId';
+    public const PARAMETER_EMPLOYEE_ID = 'employeeId';
+
+    public const PARAMETER_RULE_NAME_MAX_LENGTH = 100;
+    public const PARAMETER_RULE_JOB_TITLE_CODE_MAX_LENGTH = 4;
+    public const PARAMETER_RULE_HIRING_MANGER_ID_MAX_LENGTH = 13;
+    public const PARAMETER_RULE_NO_OF_POSITIONS_MAX_LENGTH = 13;
+    public const PARAMETER_RULE_STATUS_MAX_LENGTH = 4;
+    /**
+     * @var null|VacancyService
+     */
+    protected ?VacancyService $vacancyService = null;
+
+    /**
+     * @return EndpointResourceResult
+     * @throws DaoException
+     * @throws RecordNotFoundException
+     * @throws NormalizeException
+     */
+    public function getOne(): EndpointResourceResult
+    {
+        $id = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            CommonParams::PARAMETER_ID
+        );
+        $vacancy = $this->getVacancyService()->getVacancyById($id);
+        $this->throwRecordNotFoundExceptionIfNotExist($vacancy, Vacancy::class);
+        return new EndpointResourceResult(VacancyModel::class, $vacancy);
+    }
+
+    /**
+     * @return VacancyService
+     */
+    public function getVacancyService(): VacancyService
+    {
+        if (is_null($this->vacancyService)) {
+            $this->vacancyService = new VacancyService();
+        }
+        return $this->vacancyService;
+    }
```

Comment:
move to before `getOne`
and this getter can protected

Reply:
done